Fix issues after running static analysis: part 1

amiclaus opened the pull request:
First set of fixed issues after running cppcheck tool.

Signed-off-by: Antoniu Miclaus antoniu.miclaus@analog.com

commodo commented:
generally: each commit should have a Signed-off-by and a description about the change

amiclaus commented:
Changelog: v1->v2: 
* added "Signed-off by:" to each commit.
* added description to each commit.
* fixed commit:` ad5766-sdz: ad5766_core: fix memory leak `; removed erroneous free; moved exit path earlier in the code.
